src/static/static_context.rs

```diff
@@ -22,13 +22,24 @@ impl FutharkContext {
         }
     }
 
-    pub(crate) fn ptr(&mut self) -> *mut bindings::futhark_context {
-        self.context
+    pub(crate) fn ptr(&self) -> *mut bindings::futhark_context {
+        unsafe {
+            std::mem::transmute(self.context)
+        }
     }
 }
 
 impl From<FutharkContext> for *mut bindings::futhark_context {
-    fn from(mut ctx: FutharkContext) -> Self {
+    fn from(ctx: FutharkContext) -> Self {
```

Comment:
This should really be markes as `unsafe`. I understand the trait is neat, but it is very easy to missuse this.